### ISSUE TITLE
RegularExpressions::ProhibitUnusedCapture false positive with split

### ISSUE THREAD

**Issue by epa** [Perl-Critic/Perl-Critic#888]
This test program

```
/(.+)/;
@a = split /x/, $1;
```

saved as file `t` and checked with

```
% perlcritic --single-policy RegularExpressions::ProhibitUnusedCapture t
```

Produces the diagnostic

```
Only use a capturing group if you plan to use the captured value at line 1, column 1.  See page 252 of PBP.  (Severity: 3)
```

The code is using the captured value.  Perhaps perlcritic expects `$1` to be assigned to a named variable before use.  But if so the message should say something different.

**Comment by trwyant** [comment]
The problem here is that the policy does not realize that regexen in split() don't spit out capture variables even if they have captures. I will take a look at this, since it looks like it's my **extremely** ugly code that is at fault.

**Comment by epa** [comment]
Thanks, I can confirm that trwyant's patch (applied on top of 1.138) fixes the test case and the original example I derived it from.

**Comment by epa** [comment]
Confirmed fixed in 1.138 (if not earlier).  Thanks.
